src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -123,235 +144,205 @@ public void setShoulderJointPower(double power) {
      * @param power desired power. [-1,1]
      */
     public void setElbowJointPower(double power) {
-        elbowMainMotor.set(TalonFXControlMode.PercentOutput, power);
+        elbowMainMotor.setControl(new DutyCycleOut(power));
     }
 
-    /**
-     * Gets the angle of the shoulder joint.
-     *
-     * @return angle of the shoulder joint. [rad]
-     */
-    public Rotation2d getShoulderJointAngle() {
-        return AngleUtil.normalize(Rotation2d.fromRotations(shoulderEncoder.getAbsolutePosition() - shoulderOffset));
-    }
+
 
     /**
      * Sets the angle of the shoulder joint.
      *
      * @param angle desired angle. [degrees]
      */
     public void setShoulderJointAngle(double angle) {
-        setShoulderJointAngle(angle, 0);
+        setShoulderJointAngle(angle, 0, 0);
     }
 
-    /**
-     * Sets the angle of the shoulder joint.
-     *
-     * @param angle desired angle. [degrees]
-     */
-    public void setShoulderJointAngle(double angle, double ffMultiplier) {
+
+    public void setShoulderJointAngle(double angle, double ffMultiplier, int velocity) {
         angle = AngleUtil.normalize(angle);
-        double error = unitModelShoulder.toTicks(Math.toRadians(angle)) - unitModelShoulder.toTicks(getShoulderJointAngle().getRadians());
+        Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentShoulderAngle));
         if (shoulderEncoder.isConnected()) {
-            shoulderMainMotor.set(TalonFXControlMode.Position, shoulderMainMotor.getSelectedSensorPosition() + error,
-                    DemandType.ArbitraryFeedForward, ffMultiplier * shoulderFeedforward);
+            motorControlRequest = new PositionVoltage(angle);
+            shoulderMainMotor.setControl(new PositionVoltage(angle, velocity, true, shoulderFeedforward * ffMultiplier, 0, true));
```

Comment:
From the [Phoenix 6 documentation](https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/api-usage/control-requests.html#modifying-a-control-request):
> Java users should reuse control requests to prevent excessive invocation of the Garbage Collector.

You can create the request object as a class member and change the value as needed with [`withPosition()`](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/controls/PositionVoltage.html#withPosition(double)) (this doesn't create a new object but changes the same one).